service/src/main/java/uk/gov/hmcts/reform/fpl/service/cmo/ApproveDraftOrdersService.java

```diff
@@ -294,23 +338,36 @@ public void updateRejectedHearingOrders(Map<String, Object> data) {
             List<Element<HearingOrder>>) data.get(REFUSED_ORDERS), newArrayList());
 
         rejectedOrders.addAll(ordersToBeSent.stream()
-            .filter(bundle -> bundle.getValue().getRequestedChanges() != null)
+            .filter(bundle -> bundle.getValue().getRequestedChanges() != null
+                              && !bundle.getValue().isConfidentialOrder())
             .collect(toList()));
 
         if (!rejectedOrders.isEmpty()) {
             data.put(REFUSED_ORDERS, rejectedOrders);
         }
     }
 
-    private void updateHearingDraftOrdersBundle(CaseData caseData, Element<HearingOrdersBundle> selectedOrdersBundle) {
-        if (selectedOrdersBundle.getValue().getOrders().isEmpty()) {
-            caseData.getHearingOrdersBundlesDrafts()
-                .removeIf(bundle -> bundle.getId().equals(selectedOrdersBundle.getId()));
+    private Map<String, Object> updateHearingDraftOrdersBundle(CaseData caseData,
+                                                               Element<HearingOrdersBundle> selectedOrdersBundle) {
+        Map<String, Object> updates = new HashMap<>();
+        List<Element<HearingOrdersBundle>> hearingOrdersBundlesDrafts = caseData.getHearingOrdersBundlesDrafts();
+
+        if (selectedOrdersBundle.getValue().getOrders().isEmpty()
+                && selectedOrdersBundle.getValue().getAllConfidentialOrders().isEmpty()) {
+            hearingOrdersBundlesDrafts.removeIf(bundle -> bundle.getId().equals(selectedOrdersBundle.getId()));
+            updates.put("hearingOrdersBundlesDrafts", hearingOrdersBundlesDrafts);
         } else {
-            caseData.getHearingOrdersBundlesDrafts().stream()
+            hearingOrdersBundlesDrafts.stream()
                 .filter(bundle -> bundle.getId().equals(selectedOrdersBundle.getId()))
-                .forEach(bundle -> bundle.getValue().setOrders(selectedOrdersBundle.getValue().getOrders()));
+                .forEach(bundle -> {
+                    bundle.getValue().setOrders(selectedOrdersBundle.getValue().getOrders());
+                    selectedOrdersBundle.getValue().processAllConfidentialOrders((suffix, orders) -> {
+                        bundle.getValue().setConfidentialOrdersBySuffix(suffix, orders);
+                    });
+                });
+            updates.put("hearingOrdersBundlesDrafts", hearingOrdersBundlesDrafts);
         }
+        return updates;
```

Comment:
The return is never used.

Reply:
right....the existing code is a bit weird. I have just fixed it.